src/test/java/yevhenii/lostfilmdemo/services/impl/TVSeriesServiceImplTest.java

```diff
@@ -0,0 +1,99 @@
+package yevhenii.lostfilmdemo.services.impl;
+
+import org.jooq.Record;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.boot.test.context.SpringBootTest;
+import yevhenii.lostfilmdemo.convertors.TVRecordConvertor;
+import yevhenii.lostfilmdemo.entity.TVSeries;
+import yevhenii.lostfilmdemo.jooq.generated.tables.records.TvSeriesRecord;
+import yevhenii.lostfilmdemo.kafka.TVSeriesSender;
+import yevhenii.lostfilmdemo.repository.impl.TVSeriesRepositoryImpl;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.class)
+@SpringBootTest
+class TVSeriesServiceImplTest {
+    @Mock
+    TVSeriesRepositoryImpl repository;
+    @InjectMocks
```

Comment:
if you using mocks, which is find for your test, then you don't need to run it as integration, remove SpringBootTest annotation

Reply:
after that:


Argument passed to when() is null!
Example of correct stubbing:
    doThrow(new RuntimeException()).when(mock).someMethod();
Also, if you use @Mock annotation don't miss openMocks()
org.mockito.exceptions.misusing.NullInsteadOfMockException: 
Argument passed to when() is null!
Example of correct stubbing:
    doThrow(new RuntimeException()).when(mock).someMethod();
Also, if you use @Mock annotation don't miss openMocks()

Reply:
https://www.baeldung.com/mockito-junit-5-extension

Reply:
thx